src/util/events.ts

```diff
@@ -16,95 +17,72 @@ import { ScrollView } from './scroll-view';
  * constructor(public events: Events) {}
  * createUser(user) {
  *   console.log('User created!')
- *   events.publish('user:created', user, Date.now());
+ *   events.publish('user:created', { user, time: Date.now() });
  * }
  *
  *
  * // second page (listen for the user created event after function is called)
  * constructor(public events: Events) {
- *   events.subscribe('user:created', (user, time) => {
- *     // user and time are the same arguments passed in `events.publish(user, time)`
+ *   events.subscribe('user:created', ({ user, time }) => {
+ *     // user and time are the same arguments passed as keys in `events.publish({ ... })`
  *     console.log('Welcome', user, 'at', time);
  *   });
  * }
  *
+ * // third page utilises rxjs interface
+ * constructor(public events: Events) {
+ *   this.destroyed = new Subject<boolean>();
+ *   events.topic('user:created')
+ *     .takeUntil(this.destroyed)
+ *     .subscribe(({ user, time }) => console.log('Welcome', user, 'at', time));
+ * }
+ *
  * ```
  * @demo /docs/demos/src/events/
  */
 export class Events {
-  private _channels: any = [];
+  private _channels: { [topic: string]: Subject<any> } = {};
```

Comment:
Nothing ground breaking, I just prefer the more modern API. For instance most lint configurations I worked on will error on `delete this._channels[topic];` both for using delete and accessing an object property with brackets.
It is also more semantically correct, you are want a Map data structure and you use an object which is an approximation but not a Map.
Finally I find the reasons listed in the MDN docs under **_Objects and maps compared_** section compelling enough.

This was just a suggestion and is greatly a matter of preference, please do not feel obliged to change anything or continue this discussion.